src/streamlink/plugins/rotana.py

```diff
@@ -5,35 +5,97 @@
-    _hls_re = re.compile(r'''["'](?P<url>[^"']+\.m3u8)["']''')
+    js_url_re = re.compile(r"""['"](https://hiplayer.hibridcdn.net/l/[^'"]+)['"]""")
+    base64_data_re = re.compile(r"i\s*=\s*\[(.*)\]\.join")
 
     def _get_streams(self):
-        res = self.session.http.get(self.url)
-        m = self._hls_re.search(res.text)
-        if not m:
-            log.debug('No video URL found.')
+        js_url = self.session.http.get(
+            self.url,
+            schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'https://hiplayer.hibridcdn.net/l/')]/text()"),
+                validate.any(
+                    None,
+                    validate.all(
+                        validate.transform(self.js_url_re.search),
+                        validate.any(
+                            None,
+                            validate.all(
+                                validate.get(1),
+                                validate.url(),
+                            ),
+                        ),
+                    ),
+                ),
+            ),
+        )
+
+        if not js_url:
             return
 
-        hls_url = m.group('url')
-        log.debug('URL={0}'.format(hls_url))
-        streams = HLSStream.parse_variant_playlist(self.session, hls_url)
-        if not streams:
-            return {'live': HLSStream(self.session, hls_url)}
-        else:
-            return streams
+        log.debug(f"JS URL={js_url}")
+
+        data = self.session.http.get(
+            js_url,
+            schema=validate.Schema(
+                validate.transform(self.base64_data_re.search),
+                validate.any(
+                    None,
+                    validate.all(
+                        validate.get(1),
+                        validate.transform(lambda s: s.replace("'", "")),
+                        validate.transform(lambda s: s.replace('"', "")),
+                        validate.transform(lambda s: s.replace(",", "")),
+                        validate.transform(lambda s: s.replace(" ", "")),
+                        validate.transform(lambda s: base64.b64decode(s)),
+                        validate.parse_json(),
+                        validate.any(
+                            None,
+                            {
+                                "daiEnabled": bool,
+                                "daiAssetKey": str,
+                                "daiApiKey": str,
+                                "streamUrl": validate.any(validate.url(), ""),
+                            },
+                        ),
+                    ),
+                ),
+            ),
+        )
+
+        hls_url = data["streamUrl"]
+
+        if data["daiEnabled"]:
+            log.debug("daiEnabled=true")
+            hls_url = self.session.http.post(
+                f"https://pubads.g.doubleclick.net/ssai/event/{data['daiAssetKey']}/streams",
```

Comment:
It would probably make sense setting this JSON API URL as a class constant with a template string.